cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -309,15 +309,21 @@ func (c *Client) GetClusterVersions() (map[string]string, error) {
 
 // ClusterAvailable returns true if the cluster status indicates the cluster is available.
 func (c *Client) ClusterAvailable() (bool, error) {
-	_, err := c.GetClusterStatus()
+	_, err := c.getClusterStatus(true)
 	if err != nil {
 		return false, err
 	}
 	return true, nil
 }
 
-// GetClusterStatus returns nil for status Up or error for status Down
-func (c *Client) GetClusterStatus() (map[string]*clientv3.StatusResponse, error) {
+// CheckClusterHealth returns nil for status Up or error for status Down
+func (c *Client) CheckClusterHealth() error {
+	_, err := c.getClusterStatus(true)
+	return err
+}
+
+// getClusterStatus returns nil for status Up (along with endpoint status response map) or error for status Down
+func (c *Client) getClusterStatus(noResponseMap bool) (map[string]*clientv3.StatusResponse, error) {
```

Comment:
Being this a private func, IMO we can make this return the map and let the consumers choose if to expose to the callers or not